lightning/src/routing/scoring.rs

```diff
@@ -726,6 +733,28 @@ impl<L: DerefMut<Target = u64>, T: Time, U: DerefMut<Target = T>> DirectedChanne
 	}
 }
 
+const FRACTIONAL_BITS: u32 = 4;
+const FRACTIONAL_BITMASK: u64 = (1 << FRACTIONAL_BITS) - 1;
+const LOG2_FRACTIONAL_PART: [f64; 1 << FRACTIONAL_BITS] = [
+	0.000, 0.087, 0.170, 0.248, 0.322, 0.392, 0.459, 0.524,
+	0.585, 0.644, 0.700, 0.755, 0.807, 0.858, 0.907, 0.954,
+];
+const LOG2_10: f64 = 3.322;
+
+fn log10_approx(numerator: u64, denominator: u64) -> f64 {
+	(log2_approx(numerator) - log2_approx(denominator)) / LOG2_10
```

Comment:
The constants are only for the fractional parts, so the division would still be needed for the integer parts. Suppose I could make up a lookup table for that, too, given there would only be 64 entries. Would you prefer that?

Reply:
I guess I'm confused why we have a lookup table for log2 and then convert to log10 instead of just having a lookup table in log10, but I haven't dug too deep into how this works to begin with. In any case, a 64-entry lookup table seems more than fine too.

Reply:
The shift is used to get the exact integer part of the `log2` (i.e., most-significant bit position) and the lookup table is used to approximate the fractional part. I'm not aware of a similarly efficient approximation for `log10` though I haven't looked extensively into it.

Reply:
Actually, note that by shift I meant finding the most significant bit set. Thought that's how it was done under the hood.

Reply:
Ah, we'd probably need two integer divisions by 10 to do something similar in log10, I guess? Still probably faster than a float divide, fair enough.

Reply:
TIL this is apparently not true! Float divides can be a bit faster than int.